applications/DEMApplication/custom_utilities/inlet.cpp

```diff
@@ -310,44 +308,36 @@ namespace Kratos {
 
     void DEM_Inlet::CheckDistanceAndSetFlag(ModelPart& r_modelpart)
     {
-            std::vector<unsigned int> ElementPartition;
-            OpenMPUtils::CreatePartition(ParallelUtilities::GetNumThreads(), r_modelpart.GetCommunicator().LocalMesh().Elements().size(), ElementPartition);
-            typedef ElementsArrayType::iterator ElementIterator;
-            #pragma omp parallel
-            {
-            #pragma omp for
-            for (int k = 0; k < (int)r_modelpart.GetCommunicator().LocalMesh().Elements().size(); k++) {
-                ElementIterator elem_it = r_modelpart.GetCommunicator().LocalMesh().Elements().ptr_begin() + k;
-            if (elem_it->Is(BLOCKED)) continue;
-
-                SphericParticle& spheric_particle = dynamic_cast<SphericParticle&>(*elem_it);
-
-            if (!(*(spheric_particle.mpInlet))[DENSE_INLET]) continue;
+        ElementsArrayType& rElements = r_modelpart.GetCommunicator().LocalMesh().Elements();
+        block_for_each(rElements, [&](ModelPart::ElementType& rElement) {
+            if (rElement.Is(BLOCKED)) return;
+            SphericParticle& spheric_particle = dynamic_cast<SphericParticle&>(rElement);
+
+            if (!(*(spheric_particle.mpInlet))[DENSE_INLET]) return;
```

Comment:
I am very scared with this `return`

Reply:
We can only pull this if we are 100% sure it works. Still, the code looks ugly because the meaning of the keyword (return) is different from the usual one

Reply:
I think it should work, but I agree this should be confirmed with a small test

in any case as this mechanism is also used in the standard library (e.g. `std::find_if`) I would be ok to change it

Reply:
I will check if test_DEM_2D_inlet.py is currently going through the new code or make a new test if its not using this.

Reply:
testing with a small 3d inlet test with the DENSE_INLET flag active

Reply:
tested with #https://github.com/KratosMultiphysics/Kratos/pull/8687/commits/8acb5aab4141c5f6bb2a44b055700181fa7165f0.
seems ok and the post_files show the dense zone working as expected.